build-tools/packages/build-tools/src/fluidBuild/fluidUtils.ts

```diff
@@ -140,27 +144,41 @@ const configExplorer = cosmiconfigSync(configName, {
 	packageProp: [configName],
 });
 
+/**
+ * Set to true when the default config is returned by getFluidBuildConfig so that repeated calls to the function don't
+ * result in repeated searches for config.
+ */
+let defaultConfigLoaded = false;
+
 /**
  * Get an IFluidBuildConfig from the fluidBuild property in a package.json file, or from fluidBuild.config.[c]js.
  *
  * @param rootDir - The path to the root package.json to load.
  * @param noCache - If true, the config cache will be cleared and the config will be reloaded.
- * @returns The fluidBuild section of the package.json, or undefined if not found
+ * @returns The the loaded fluidBuild config, or the default config if one is not found.
  */
 export function getFluidBuildConfig(
 	rootDir: string,
 	noCache = false,
 	log = defaultLogger,
 ): IFluidBuildConfig {
+	if (defaultConfigLoaded) {
+		return DEFAULT_FLUIDBUILD_CONFIG;
+	}
+
 	if (noCache === true) {
 		configExplorer.clearCaches();
 	}
 
 	const configResult = configExplorer.search(rootDir);
-	const config = configResult?.config as IFluidBuildConfig | undefined;
+	const config: IFluidBuildConfig = configResult?.config ?? DEFAULT_FLUIDBUILD_CONFIG;
```

Comment:
I think this line wants to be after the `if` below and won't need the coalescing anymore.

Reply:
It can, but then config has to be mutable, which I was trying to avoid. But given that this stuck out to you, it will
likely cause others to do a double-take, so I'll reorg it to be more typical.

Reply:
Refactored to remove the need for the mutable variable.